adapter/slack.py

```diff
@@ -51,4 +53,27 @@ async def ask(self, request: Request):
                     raise HTTPException(status_code=400, detail=f"Slack API Error : {delete_error}")
 
             raise HTTPException(status_code=400, detail=f"Slack API Error : {e}")
+    
+    async def list_bots(self, request: Request):
+        data = await request.form()
+    
+        channel_id = data.get("channel_id")
+        
+        bot_responses = self.bot_controller.fetch_chatbots()
         
+        response_text = f"{len(bot_responses)} Active Bots:"
+        response_text += ''.join([f'\n- {bot_response.name} ({bot_response.id})' for bot_response in bot_responses])
+        
+        try:
+            response = self.app.client.chat_postMessage(channel=channel_id, text=response_text)

```

Comment:
I don't think you need to store the return value from chat_postMessage or save the thread_ts. In the ask method, this is necessary because there will be a feature to reply to the thread.

Reply:
thanks for the clarification, will remove the variable and other unused code.

Reply:
Have just removed the irrelevant codes!